### DISCUSSION TITLE
Current approach for Scoping State?

### DISCUSSION THREAD

**Discussion by opsb** [GregoryConrad/rearch-dart#192] [Q&A] [answered]
The example at https://rearch.gsconrad.com/flutter/scoping-state shows 

```dart
(int, void Function()) scopedCount(WidgetHandle use, int startingCount) {
  final (count, setCount) = use.state(startingCount);
  return (count, () => setCount(count + 1));
}
```

but currently WidgetHandle doesn't appear to have the `use.state` (or `use.data`). Is there an approach to achieve the same with the current API?

**Comment by GregoryConrad** [comment, marked as answer]
Those are defined as extension methods over in `rearch`; thus:

`import 'package:rearch/rearch.dart'; `

**Reply by opsb** [reply]
Ah yep, I just discovered that. It seems vscode isn't able to discover the import.
